code/modules/fishing/fish/types/rift.dm

```diff
@@ -0,0 +1,394 @@
+	drop_sound = 'sound/items/handling/materials/glass_drop.ogg'
+
+	sprite_width = 7
+	sprite_height = 9
+
+	average_size = 40
+	average_weight = 1500
+	food = /datum/reagent/bluespace
+	feeding_frequency = 10 MINUTES
+	health = 50
+	death_text = "%SRC splinters apart into shards!"
+	random_case_rarity = FISH_RARITY_GOOD_LUCK_FINDING_THIS
+	fillet_type = /obj/item/stack/ore/bluespace_crystal
+	num_fillets = 3
+	stable_population = 3
+	compatible_types = list(/obj/item/fish/starfish)
+	fish_traits = list(/datum/fish_trait/antigrav)
+	beauty = FISH_BEAUTY_EXCELLENT
+	fish_movement_type = /datum/fish_movement/accelerando
+	fishing_difficulty_modifier = 15
+	favorite_bait = list(/obj/item/stack/ore/bluespace_crystal)
+	// something something bluespace
+	electrogenesis_power = 9 MEGA JOULES
+
+// Basically a ninja star that's highly likely to embed and teleports you around if you don't stop to remove it. However it doesn't deal that much damage!
+/datum/embed_data/chrystarfish
+	pain_mult = 1
+	embed_chance = 85
+	fall_chance = 3
+	pain_chance = 9
+	impact_pain_mult = 1
+	remove_pain_mult = 2
+	rip_time = 1.5 SECONDS
+	ignore_throwspeed_threshold = TRUE // basically shaped like a shuriken
+	jostle_chance = 15
+	jostle_pain_mult = 1
+	// about to be set!
+	jostle_callback = null
+
+/datum/embed_data/chrystarfish/New()
+	..()
+	jostle_callback = CALLBACK(src, PROC_REF(teleport))
+
+/datum/embed_data/chrystarfish/proc/teleport(mob/victim, atom/embed_parent, datum/embed_data/real_data)
+	do_teleport(victim, get_turf(victim), 3, asoundin = 'sound/effects/phasein.ogg', channel = TELEPORT_CHANNEL_BLUESPACE)
+	victim.visible_message(span_danger("[victim] teleports as [embed_parent] jostles inside [victim.p_them()]!"))
+
+/obj/item/fish/starfish/chrystarfish/set_status(new_status, silent)
+	. = ..()
+	if(new_status == FISH_DEAD)
+		new fillet_type(get_turf(src))
+		playsound(src, SFX_SHATTER, 50)
+		qdel(src)
+
+/obj/item/fish/starfish/chrystarfish/add_emissive()
+	return
+
+/obj/item/fish/starfish/chrystarfish/get_base_edible_reagents_to_add()
+	var/list/return_list = ..()
+	return_list[/datum/reagent/bluespace] = 5
+	return return_list
+
+/obj/item/fish/starfish/chrystarfish/flinch_on_eat(mob/living/eater, mob/living/feeder)
+	if(status != FISH_ALIVE)
+		return
+	to_chat(feeder, span_warning("[src] slips out of the spacetime in pain!"))
+
+	var/tp_range = 6 * clamp(weight/average_weight, 3, 9) // usually 6, plus or minus fish weight
+	// teleports itself if on a turf otherwise its container - whatever it is
+	do_teleport(isturf(loc) ? src : loc, get_turf(feeder), tp_range, asoundin = 'sound/effects/phasein.ogg', channel = TELEPORT_CHANNEL_BLUESPACE)
+
+/obj/item/fish/starfish/chrystarfish/suicide_act(mob/living/user)
+	visible_message(span_suicide("[user] swallows [src] whole! It looks like they're trying to commit suicide!"))
+	forceMove(user)
+	// *everything*
+	for(var/obj/thing in user.get_contents())
+		if(istype(thing, /obj/item/bodypart/chest))
```

Comment:
```suggestion
		if(QDELETED(thing) || istype(thing, /obj/item/bodypart/chest))
```
It's entirely possible that some objects are deleted during the loop as a reaction of other objects being moved.